[Bug]: I was trying to create an aws_config_rule, cloud_watch event & cloud_watch_target. But the provider was crashing when creating config_rules. It's working fine from my local but failing when running using gitlab-ci pipeline

### Terraform Core Version

Terraform v1.3.6 on linux_amd64

### AWS Provider Version

- Finding hashicorp/aws versions matching "~> 4.37"... - Installing hashicorp/aws v4.67.0... - Installed hashicorp/aws v4.67.0 

### Affected Resource(s)

aws_config_rule

### Expected Behavior

It should create rule and then create cloudwatch event & cloud_watch target

### Actual Behavior

```shell
Error: Plugin did not respond
│ 
│   with module.kms_rotation_check_west.aws_config_config_rule.cmk-backing-key-rotation-enabled,
│   on modules/kms-key-rotation-check/main.tf line 11, in resource "aws_config_config_rule" "cmk-backing-key-rotation-enabled":
│   11: resource "aws_config_config_rule" "cmk-backing-key-rotation-enabled" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Request cancelled
│ 
│   with module.kms_rotation_check_east.aws_config_config_rule.cmk-backing-key-rotation-enabled,
│   on modules/kms-key-rotation-check/main.tf line 11, in resource "aws_config_config_rule" "cmk-backing-key-rotation-enabled":
│   11: resource "aws_config_config_rule" "cmk-backing-key-rotation-enabled" {
│ 
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵
Stack trace from the terraform-provider-aws_v4.67.0_x5 plugin:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x96a0818]
goroutine 392 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/configservice.resourceConfigRuleRead({0xf0a19c0, 0xc00609f0e0}, 0xc00584a700, {0xdb8cd20?, 0xc0003dc800?})
	github.com/hashicorp/terraform-provider-aws/internal/service/configservice/config_rule.go:252 +0x258
github.com/hashicorp/terraform-provider-aws/internal/service/configservice.resourceRulePutConfig({0xf0a19c0?, 0xc00609f0e0}, 0x1?, {0xdb8cd20?, 0xc0003dc800})
	github.com/hashicorp/terraform-provider-aws/internal/service/configservice/config_rule.go:237 +0x712
github.com/hashicorp/terraform-provider-aws/internal/provider.interceptedHandler[...].func1(0x0?, {0xdb8cd20?, 0xc0003dc800?})
	github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:95 +0x175
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xf0a19c0?, {0xf0a19c0?, 0xc006098540?}, 0xd?, {0xdb8cd20?, 0xc0003dc800?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:702 +0x84
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000d9fa40, {0xf0a19c0, 0xc006098540}, 0xc005844dd0, 0xc00584a580, {0xdb8cd20, 0xc0003dc800})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:837 +0xa85
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc001617698, {0xf0a19c0?, 0xc006098420?}, 0xc0033b1090)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/grpc_provider.go:1021 +0xe8d
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ApplyResourceChange({0xc00374d4a0, 0xc00374d500, {0xc0065d5740, 0x2, 0x2}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
	github.com/hashicorp/terraform-plugin-mux@v0.10.0/tf5muxserver/mux_server_ApplyResourceChange.go:27 +0x102
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0032530e0, {0xf0a19c0?, 0xc006070ba0?}, 0xc005403e30)
	github.com/hashicorp/terraform-plugin-go@v0.15.0/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xd877e80?, 0xc0032530e0}, {0xf0a19c0, 0xc006070ba0}, 0xc005403dc0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.15.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:419 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc005fc6780, {0xf0b21e0, 0xc0005a0680}, 0xc005846360, 0xc0066df7a0, 0x15cfa500, 0x0)
	google.golang.org/grpc@v1.54.0/server.go:[134](https://xxx/-/jobs/6713003#L134)5 +0xdf0
google.golang.org/grpc.(*Server).handleStream(0xc005fc6780, {0xf0b21e0, 0xc0005a0680}, 0xc005846360, 0x0)
	google.golang.org/grpc@v1.54.0/server.go:1722 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.54.0/server.go:966 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.54.0/server.go:964 +0x28a
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x96a0818]
goroutine 412 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/configservice.resourceConfigRuleRead({0xf0a19c0, 0xc0051957d0}, 0xc004a22600, {0xdb8cd20?, 0xc001903000?})
	github.com/hashicorp/terraform-provider-aws/internal/service/configservice/config_rule.go:252 +0x258
github.com/hashicorp/terraform-provider-aws/internal/service/configservice.resourceRulePutConfig({0xf0a19c0?, 0xc0051957d0}, 0x1?, {0xdb8cd20?, 0xc001903000})
	github.com/hashicorp/terraform-provider-aws/internal/service/configservice/config_rule.go:237 +0x712
github.com/hashicorp/terraform-provider-aws/internal/provider.interceptedHandler[...].func1(0x0?, {0xdb8cd20?, 0xc001903000?})
	github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:95 +0x175
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xf0a19c0?, {0xf0a19c0?, 0xc0051643c0?}, 0xd?, {0xdb8cd20?, 0xc001903000?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:702 +0x84
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000d67a40, {0xf0a19c0, 0xc0051643c0}, 0xc004a01860, 0xc004a22480, {0xdb8cd20, 0xc001903000})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:837 +0xa85
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc001794b40, {0xf0a19c0?, 0xc005161bf0?}, 0xc00318ecd0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/grpc_provider.go:1021 +0xe8d
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ApplyResourceChange({0xc0042a5f80, 0xc0042cc000, {0xc006349d40, 0x2, 0x2}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
	github.com/hashicorp/terraform-plugin-mux@v0.10.0/tf5muxserver/mux_server_ApplyResourceChange.go:27 +0x102
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc002cd5040, {0xf0a19c0?, 0xc0046470b0?}, 0xc00272d7a0)
	github.com/hashicorp/terraform-plugin-go@v0.15.0/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xd877e80?, 0xc002cd5040}, {0xf0a19c0, 0xc0046470b0}, 0xc00272d730, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.15.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:419 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc005de85a0, {0xf0b21e0, 0xc004d0a680}, 0xc004a1e000, 0xc006476bd0, 0x15cfa500, 0x0)
	google.golang.org/grpc@v1.54.0/server.go:1345 +0xdf0
google.golang.org/grpc.(*Server).handleStream(0xc005de85a0, {0xf0b21e0, 0xc004d0a680}, 0xc004a1e000, 0x0)
	google.golang.org/grpc@v1.54.0/server.go:1722 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.54.0/server.go:966 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.54.0/server.go:964 +0x28a
Error: The terraform-provider-aws_v4.67.0_x5 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
Cleaning up project directory and file based variables
00:01
ERROR: Job failed: command terminated with exit code 1
```

### Relevant Error/Panic Output Snippet

_No response_

### Terraform Configuration Files

```terraform
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 4.37"
    }
  }
}


resource "aws_config_config_rule" "cmk-backing-key-rotation-enabled" {
  name        = "${var.env}-${var.region}-kmskey-rotationenabledcheck"
  description = "Checks if automatic key rotation is enabled for each key and matches to the key ID of the customer created AWS KMS key. The rule is NON_COMPLIANT if the AWS Config recorder role for a resource does not have the kms:DescribeKey permission."

  source {
    owner             = "AWS"
    source_identifier = "CMK_BACKING_KEY_ROTATION_ENABLED"
  }

  maximum_execution_frequency = "One_Hour"
}

resource "aws_cloudwatch_event_rule" "cmk-backing-key-rotation-enabled" {
  name        = "${var.env}-${var.region}-kmskey-rotationenabledcheck"
  description = "Rule to detect NON_COMPLIANT resources for kms_keys"

  event_pattern = jsonencode({

    source      = ["aws.config"],
    detail-type = ["Config Rules Compliance Change"],
    detail = {
      messageType    = ["ComplianceChangeNotification"],
      configRuleName = ["cmk-backing-key-rotation-enabled"],
      resourceType   = ["AWS::KMS::Key"],
      newEvaluationResult = {
        complianceType = ["NON_COMPLIANT"]
      }
    }
  })
  depends_on = [aws_config_config_rule.cmk-backing-key-rotation-enabled]
}

resource "aws_cloudwatch_event_target" "cmk-backing-key-rotation-enabled" {
  rule      = aws_cloudwatch_event_rule.cmk-backing-key-rotation-enabled.name
  target_id = "CheckForKMSKeysRotationEnabled"
  arn       = var.aws_sns_topic

  input_transformer {
    input_paths = {
      awsRegion    = "$.detail.awsRegion",
      resourceId   = "$.detail.resourceId",
      awsAccountId = "$.detail.awsAccountId",
      compliance   = "$.detail.newEvaluationResult.complianceType",
      rule         = "$.detail.configRuleName",
      time         = "$.detail.newEvaluationResult.resultRecordedTime",
      resourceType = "$.detail.resourceType"
    }

    input_template = "\"On <time> AWS Config rule <rule> evaluated the <resourceType> with Id <resourceId> in the account <awsAccountId> region <awsRegion> as <compliance>. For more details, open the AWS Config console at https://console.aws.amazon.com/config/home?region=<awsRegion>#/timeline/<resourceType>/<resourceId>/configuration\""
  }

}
```

### Steps to Reproduce

run terraform apply

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

None

--- a/.changelog/32439.txt
+++ b/.changelog/32439.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_config_config_rule: Prevent crash on nil describe output
+```

--- a/internal/service/configservice/find.go
+++ b/internal/service/configservice/find.go
@@ -26,10 +26,6 @@ func FindConfigRule(ctx context.Context, conn *configservice.ConfigService, name
 		}
 	}
 
-	if output == nil {
-		return nil, nil
-	}
-
 	if output == nil || output.ConfigRules == nil || len(output.ConfigRules) == 0 || output.ConfigRules[0] == nil {
 		return nil, tfresource.NewEmptyResultError(input)
 	}

